client/src/components/SignatureList.js

```diff
@@ -1,25 +1,48 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { fetchCampaignSignatures } from '../redux/actions/signature';
 
-const SignatureList = ({ signatures }) => (
-  <div>
-    {signatures.length ? (
-      <h2>
-        ({signatures.length}) Signature{signatures.length > 1 ? 's' : ''} on the list!
-      </h2>
-    ) : (
-      <h2>Be the First!</h2>
-    )}
-    <ul>{signatures && signatures.map(({ id, name }) => <li key={id}>{name}</li>)}</ul>
-  </div>
-);
+class SignatureList extends React.Component {
+  componentDidMount() {
+    this.props.fetchCampaignSignatures(this.props.campaignID);
+  }
+
+  componentWillUpdate(nextProps, nextState) {
+    if (this.props.campaignID !== nextProps.campaignID) {
+      nextProps.fetchCampaignSignatures(nextProps.campaignID);
+    }
+  }
+
+  render() {
+    const { signatures } = this.props;
+    return (
+      <div>
+        {signatures.length ? (
+          <h2>
+            ({signatures.length}) Signature{signatures.length > 1 ? 's' : ''} on the list!
+          </h2>
+        ) : (
+          <h2>Be the First!</h2>
+        )}
+        <ul>{signatures && signatures.map(({ id, name }) => <li key={id}>{name}</li>)}</ul>
+      </div>
+    );
+  }
+}
 
 SignatureList.defaultProps = {
   signatures: []
 };
 
 SignatureList.propTypes = {
-  signatures: PropTypes.arrayOf(PropTypes.object)
+  campaignID: PropTypes.string.isRequired,
+  signatures: PropTypes.arrayOf(PropTypes.object),
+  fetchCampaignSignatures: PropTypes.func.isRequired
 };
 
-export default SignatureList;
+const mapStateToProps = state => ({
```

Comment:
Same as above

Reply:
since mapStateToProps is the more explicit, and more documented online, I think it would be good to push to that direction over time in our app.